### ISSUE TITLE
[Proposal] "Other" Charge Subcategory for usage should not be the default

### ISSUE THREAD

**Issue by flanakin** [FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec#305]
### Type
> Dimension: ChargeSubcategory
> Normalized: Yes

Description when ChargeCategory = "Usage":
> _When Charge Category is "Usage" and the charge is related to a commitment, the Charge Subcategory indicates whether the row represents committed usage or is an amortized charge for the unused portion of the commitment. Charge Subcategory is commonly used for scenarios like calculating commitment utilization when Charge Category is "Usage"._

### Description
> ChargeSubcategory for usage allows values "Used Commitment", "Unused Commitment", and "Other". This means that all on-demand (non-committed) usage is shown as "Other". Given this is the most common scenario, the feedback I've heard is that "Other" is not a good value to use to represent on-demand usage.
>
> I would propose we either use "On-Demand" or consider other alternative ways to represent ChargeSubcategory for usage.

### Definition of done
- [x] Rationalize vendor-neutral, cross-cloud naming
- [x] Complete spec template and include naming (code name, display name), constraints, guidelines, compatibility with major providers etc.
- [x] Include principles and governance criteria for maintaining this dimension

**If Normalized Dimension**
Work for generating the normalized list of supported values is tracked in a separate issue. Mappings between normalized values and vendor specified values need to be explored as a part of this work. However, these mappings are not included in the spec documentation. Separate tasks will be created for making these mappings available to practitioners outside of the FOCUS repository.

<br>

### Give it a 👍 below ↴

If you can discuss and help finalize the change, add yourself as an assignee ↗

Comments are welcome and encouraged!

**Comment by mike-finopsorg** [comment]
+1 to remove the over use of `other`. I think that using `On-Demand` while has some overlap is the most commonly understood and adds value over `other`.

**Comment by marc-perreaut** [comment]
I probably miss something, but the spec says:

> ChargeSubcategory MUST be null when ChargeCategory is "Usage" and is not covered by a *commitment*.

https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/blob/working_draft/specification/columns/chargesubcategory.md?plain=1#L18

**Comment by ijurica** [comment]
Few more suggestions/questions:

- Consider mandating the use of SkuPriceId for Refunds and stipulating that bulk refunds be categorized as General Adjustments
- Consider splitting Credit in two ChargeSubcategories: one for UCs when SkuPriceId MUST be null and the other for UCs when it MUST NOT be null

Reminder:
- As negotiated credits are categorized as credits, it's important to consider/identify those charges when calculating savings based on negotiation activities.

**Comment by flanakin** [comment]
> I probably miss something, but the spec says:
> 
> > ChargeSubcategory MUST be null when ChargeCategory is "Usage" and is not covered by a _commitment_.

🤦‍♂️ You're right. I think I got confused with all the conversations. At one point, we talked about using "Other" and I apparently lost track of where we landed.

I proposed closing this issue, but we talked about repurposing this for the general commitment discount and ChargeSubcategory structure.

**Comment by flanakin** [comment]
> * Consider mandating the use of SkuPriceId for Refunds and stipulating that bulk refunds be categorized as General Adjustments
> * Consider splitting Credit in two ChargeSubcategories: one for UCs when SkuPriceId MUST be null and the other for UCs when it MUST NOT be null

These would means there wouldn't be a single filter for all refunds or all credits. As-is, all refunds are `ChargeSubcategory == "Refund"` and bulk refunds are `ChargeSubcategory == "Refund" and SkuPriceId == null`. Same for credits.

I understand the value in a validator to check those, but I suspect that would just push people to always use the bulk option. If that happens, then there's no value in splitting them and it's just harder for practitioners.

This reminds me that we also talked about changing "General Adjustments" to "Other" for consistency with other normalized columns.

> As negotiated credits are categorized as credits, it's important to consider/identify those charges when calculating savings based on negotiation activities.

I'm not following this one. But if there are different types of credits or refunds, then we just need to determine whether we want them grouped together or not. We could either have separate ChargeSubcategory values or have another level, if it's really needed.

**Comment by ijurica** [comment]
Clarification on the comment regarding Credits and savings resulting from negotiation activities...

In light of the current definition of the Credit ChargeSubcategory:

>Credit: Negative charges granted by the provider for various scenarios, like negotiated benefits, usage discounts, or promotional credits.

- **Credits-related charges based on negotiated benefits:** If ensuring/verifying that all negotiated benefits are realized and/or calculating negotiation-based savings are important UCs, we need to be able to identify those particular charges.

- **Credits-related charges based on usage discounts:** When should we expect Credits rather than Usage charges with tier/volume-based pricing?

**Comment by flanakin** [comment]
> If ensuring/verifying that all negotiated benefits are realized and/or calculating negotiation-based savings are important UCs, we need to be able to identify those particular charges.

Ah, yes. Totally agree there. But doesn't that apply to every charge? While we could split out a credit into multiple rows with different ChargeSubcategory values, that doesn't help us in the other columns. Shouldn't we evaluate identification of negotiated discounts holistically? I would expect that however we do that plus `ChargeSubcategory == "Credit"` would give us the answer, but that depends on how we solve it.

**Comment by flanakin** [comment]
> When should we expect Credits rather than Usage charges with tier/volume-based pricing?

Similar thought... Do we need to define tiered usage before we can answer this? I haven't put any thought into this explicitly. I suppose this might be simpler since you're not talking about splitting "Credit" necessarily and just talking about being more explicit about how we define "Credit". (Of course, we may decide to split it in the end.)

**Comment by udam-f2** [comment]
If we need to show on-demand spending, we should have a value for it. If we need to differentiate committable spend vs spend not eligible for commitment differently, we could go
On-Demand
Commitment Eligible
Used Commitment
Unused Commitment

The downside is we don't have single responsibility columns so people would have to sum multiple values to find the total non-committable spend OR total committed spend etc. but we're already in that realm with the way things are set up anyway. If we have this, we won't need an 'NULL' or 'Other' value for ChargeType 'Usage'. Thoughts?

**Comment by flanakin** [comment]
Consolidating this with other charge/pricing categorization issues to discuss together.

@ijurica / @ahullah please move 👍 votes to #330
